### ISSUE TITLE
[AVM Question/Feedback]:

### ISSUE THREAD

**Issue by zioproto** [Azure/terraform-azurerm-avm-res-containerservice-managedcluster#13]
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Description

I am looking at the `cluster_suffix` variable:

https://github.com/Azure/terraform-azurerm-avm-res-containerservice-managedcluster/blob/9bb85205064513f788b1995f692715dc594a3109/variables.tf#L180-L184

I dont understand why we need it.

The only place where this variable is used is here:
https://github.com/Azure/terraform-azurerm-avm-res-containerservice-managedcluster/blob/9bb85205064513f788b1995f692715dc594a3109/main.tf#L1-L4

Can't we just use a different name ?

I also looked at the `create-before-destroy` example and the variable is not used. 

https://github.com/Azure/terraform-azurerm-avm-res-containerservice-managedcluster/blob/9bb85205064513f788b1995f692715dc594a3109/examples/create-before-destroy/main.tf#L52-L110

I don't understand the description saying: "the suffix to append to the Kubernetes cluster name if create_before_destroy is set to true on the nodepools."
Why this static suffix is needed ?
thank you

**Comment by ibersanoMS** [comment]
Yes, that was the idea or they have some automation that generates a new suffix each time it deploys. They could just change the name and we make it a requirement to change the name if they enable `create_before_destroy`. We can discuss changing that in an upcoming release
